### ISSUE TITLE
[unicorn/no-useless-spread] False-positive with Array constructor

### ISSUE THREAD

**Issue by MorevM** [sindresorhus/eslint-plugin-unicorn#1994]
Hi, usage of spread is necessary if you create an array via Array constructor and you need to iterate over it.
```js
[...new Array(10)]; // Array of 10 `undefined` elements (iterable)
new Array(10);      // Array of 10 `empty` elements (non-iterable)
```

Buggy rule is [unicorn/no-useless-spread](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v45.0.0/docs/rules/no-useless-spread.md)

I personally face the issue [here](https://github.com/MorevM/helpers/blob/748c5be9f16ee794c78d18d285447254b7145691/src/strings/random-string/random-string.ts#L13)

```js
// Results in a random 10-char string 
[...new Array(10)].map(() => (Math.trunc(Math.random() * 36)).toString(36)).join('');

// Results in an empty string
new Array(10).map(() => (Math.trunc(Math.random() * 36)).toString(36)).join('');
```

Workaround is to use `.fill()` or disable the rule temporarly.
```js
// Works the same as when using spread - 10-char random string
new Array(10).fill().map(() => (Math.trunc(Math.random() * 36)).toString(36)).join('');
```

**Comment by fisker** [comment]
I agree we should skip auto-fix for this case.

But I'd use

`Array.from({length: 10}, () => (Math.trunc(Math.random() * 36)).toString(36)).join(''))`

**Comment by MorevM** [comment]
Feels like personal stylistic preferences :)

I did a [performance test for both variants](https://www.measurethat.net/Benchmarks/Show/22279/0/spread-vs-arrayfrom-performance) and funny thing is [Array.from is faster using Firefox](https://www.measurethat.net/Benchmarks/ShowResult/359614), but [in Chrome spread operator faster](https://www.measurethat.net/Benchmarks/ShowResult/359613).

**Comment by fisker** [comment]
See [`no-new-array`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-new-array.md)

**Comment by fisker** [comment]
I'm going to remove the auto-fix. You still need use `eslint-disable` for this case.

**Comment by MorevM** [comment]
Mention [no-new-array](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-new-array.md) makes sense to me.
I'll review my code to use `Array.from({ length: 10 })` since it's more unambiguous. Thanks.
